lib/build.ts

```diff
@@ -20,13 +20,19 @@ import * as _ from 'lodash';
 import { Builder, BuildHooks, FromTagInfo } from 'resin-docker-build';
 import * as semver from 'semver';
 import type * as Stream from 'stream';
+import { DockerfileParser } from 'dockerfile-ast';
```

Comment:
`DockerfileParser` seems to be used only inside function `getRecommendedPlatformhHandling`, and even then only in a certain decision branch, so I suggest moving the import clause to the line immediately before the point where it is used:
```
    const { DockerfileParser } = await import('dockerfile-ast')
    const parsedDockerfile = DockerfileParser.parse(task.dockerfile);
```

You will find inconsistencies in the codebase, but in general we prefer to do this because it improves "responsiveness" (performance -- time between typing the command and getting some response), especially when there are code execution paths that might no need the module. For example, `if (task.imageName)` is true, then there is no need for the CLI to spend time parsing and importing `'dockerfile-ast'`, as it won't be used. This is generally not an issue in long-running web applications (like balena-builder), but it's an issue in interactive applications like the balena CLI. While a single import clause doesn't mean much, it adds up when this approach is followed throughout the codebase.